lib/rspec/mocks/matchers/receive.rb

```diff
@@ -55,8 +55,9 @@ def setup_any_instance_allowance(subject, &block)
           setup_any_instance_method_substitute(subject, :stub, block)
         end
 
+        own_methods = (instance_methods - superclass.instance_methods)
```

Comment:
Note, I used `superclass.instance_methods` here, but depending on what the exact intent was, `Object.instance_methods` may make more sense.